pg_es_fdw/_es_query.py

```diff
@@ -65,14 +73,47 @@ def _qual_to_es(qual, column_map=None):
             }
         }
     else:
-        return _base_qual_to_es(
-            qual.field_name, qual.operator, qual.value, column_map
-        )
+        return _base_qual_to_es(qual.field_name, qual.operator, qual.value, column_map)
 
 
-def quals_to_es(quals, ignore_columns=None, column_map=None):
+def quals_to_es(
+    quals, aggs=None, group_clauses=None, ignore_columns=None, column_map=None
+):
     """Convert a list of Multicorn quals to an ElasticSearch query"""
     ignore_columns = ignore_columns or []
+
+    # Aggreagtion/grouping queries
+    if aggs is not None:
+        aggs_query = {
+            agg_name: {
+                _PG_TO_ES_AGG_FUNCS[agg_props["function"]]: {
+                    "field": agg_props["column"]
+                }
+            }
+            for agg_name, agg_props in aggs.items()
+        }
+
+        if group_clauses is None:
+            return {"aggs": aggs_query}
+
+    if group_clauses is not None:
+        group_query = {
+            "group_buckets": {
+                "composite": {
+                    "sources": [
+                        {column: {"terms": {"field": column}}}
+                        for column in group_clauses
+                    ]
+                }
+            }
+        }
+
+        if aggs is not None:
```

Comment:
Can we be in a situation where aggs is None and group_clauses isn't? Is it basically something like `SELECT a, b, c FROM T GROUP BY a, b, c` which is the same as `SELECT DISTINCT a, b, c FROM T`?

Reply:
Yes we can, that is a good example. Here's a concrete one:
```sql
sgr@localhost:splitgraph> explain select column5, column4 from es.iris group by column4, column5
+--------------------------------------------------------------------------------------------+
| QUERY PLAN                                                                                 |
|--------------------------------------------------------------------------------------------|
| Foreign Scan  (cost=1.00..1.00 rows=1 width=1)                                             |
|   Multicorn: Elasticsearch query to <Elasticsearch([{'host': 'es01-test', 'port': 9200}])> |
|   Multicorn: Query: {                                                                      |
|     "aggs": {                                                                              |
|         "group_buckets": {                                                                 |
|             "composite": {                                                                 |
|                 "sources": [                                                               |
|                     {                                                                      |
|                         "column5": {                                                       |
|                             "terms": {                                                     |
|                                 "field": "column5"                                         |
|                             }                                                              |
|                         }                                                                  |
|                     },                                                                     |
|                     {                                                                      |
|                         "column4": {                                                       |
|                             "terms": {                                                     |
|                                 "field": "column4"                                         |
|                             }                                                              |
|                         }                                                                  |
|                     }                                                                      |
|                 ],                                                                         |
|                 "size": 1000                                                               |
|             }                                                                              |
|         }                                                                                  |
|     }                                                                                      |
| }                                                                                          |
+--------------------------------------------------------------------------------------------+
EXPLAIN
Time: 0.012s
sgr@localhost:splitgraph> select column5, column4 from es.iris group by column4, column5
+-----------------+---------+
| column5         | column4 |
|-----------------+---------|
| Iris-setosa     | 0.1     |
| Iris-setosa     | 0.2     |
| Iris-setosa     | 0.3     |
| Iris-setosa     | 0.4     |
| Iris-setosa     | 0.5     |
| Iris-setosa     | 0.6     |
| Iris-versicolor | 1.0     |
| Iris-versicolor | 1.1     |
| Iris-versicolor | 1.2     |
| Iris-versicolor | 1.3     |
| Iris-versicolor | 1.4     |
| Iris-versicolor | 1.5     |
| Iris-versicolor | 1.6     |
| Iris-versicolor | 1.7     |
| Iris-versicolor | 1.8     |
| Iris-virginica  | 1.4     |
| Iris-virginica  | 1.5     |
| Iris-virginica  | 1.6     |
| Iris-virginica  | 1.7     |
| Iris-virginica  | 1.8     |
| Iris-virginica  | 1.9     |
| Iris-virginica  | 2.0     |
| Iris-virginica  | 2.1     |
| Iris-virginica  | 2.2     |
| Iris-virginica  | 2.3     |
| Iris-virginica  | 2.4     |
| Iris-virginica  | 2.5     |
+-----------------+---------+
```